app/views/questionnaire.py

```diff
@@ -199,6 +215,50 @@ def get_thank_you():
     return _redirect_to_latest_location(routing_path, collection_id, metadata.get('eq_id'), metadata.get('form_type'))
 
 
+@post_submission_blueprint.route('/view-submission', methods=['GET'])
+@login_required
+def get_view_submission():  # pylint: disable=unused-argument
+
+    session_data = get_session_store().session_data
+    submission_valid_until = int(dateutil.parser.parse(session_data.submitted_time).strftime('%s')) \
+        + session_data.submission_validity
+
+    if submission_valid_until > int(datetime.utcnow().strftime('%s')):
+
+        submitted_data = SubmittedResponses().get_item(session_data.tx_id)
+
+        if submitted_data and submitted_data['valid_until'] > int(datetime.utcnow().strftime('%s')):
+
+            metadata_context = build_metadata_context_for_survey_completed(vars(session_data))
+
+            current_location = Location('group', 0, 'view-submission')
```

Comment:
This code smells a bit. Why do we have a hard-coded Location?

Reply:
The only thing that this location is used for is to go into `_get_schema_context()` and that only uses the `group_instance` so it might be worth refactoring `_get_schema_context()` to just take in the `group_instance`